spec/compiler/semantic/abstract_def_spec.cr

```diff
@@ -394,4 +394,165 @@ describe "Semantic: abstract def" do
       end
       )
   end
+
+  it "errors if missing return type" do
+    assert_error %(
+      abstract class Foo
+        abstract def foo : Int32
+      end
+
+      class Bar < Foo
+        def foo
+          1
+        end
+      end
+      ),
+      "this method overrides Foo#foo() which has an explicit return type of Int32.\n\nPlease add an explicit return type (Int32 or a subtype of it) to this method as well."
+  end
+
+  it "errors if different return type" do
+    assert_error %(
+      abstract class Foo
+        abstract def foo : Int32
+      end
+
+      class Bar < Foo
+        struct Int32
+        end
+
+        def foo : Int32
+          1
+        end
+      end
+      ),
+      "this method must return Int32, which is the return type of the overwritten method Foo#foo(), or a subtype of it, not Bar::Int32"
```

Comment:
Should this be `overridden` as opposed to `overwritten`?

Reply:
Also here: https://github.com/crystal-lang/crystal/pull/7956/files#diff-289f0271db3c680bd06314a3ed31eae4R186

Reply:
Yes, thanks!